models/location_helpers.go

```diff
@@ -274,6 +278,75 @@ func ChangeBackupLocation(q *reform.Querier, locationID string, params ChangeBac
 	return row, nil
 }
 
+func testS3Config(c *S3LocationConfig) error {
+	parsedURL, err := url.Parse(c.Endpoint)
+	if err != nil {
+		return status.Errorf(codes.InvalidArgument, "%s", err)
+	}
+
+	endpoint := path.Join(parsedURL.Host, parsedURL.Path)
+	secure := true
+	if parsedURL.Scheme == "http" {
+		secure = false
+	}
+	minioClient, err := minio.New(endpoint, c.AccessKey, c.SecretKey, secure)
+	if err != nil {
+		return status.Errorf(codes.InvalidArgument, "%s", err)
+	}
+
+	exists, err := minioClient.BucketExists(c.BucketName)
+	if err != nil {
+		if er, ok := err.(minio.ErrorResponse); ok {
+			return status.Errorf(codes.InvalidArgument, "%s: %s.", er.Code, er.Message)
```

Comment:
Why invalid argument?

Reply:
when mini.ErrorResponse is returned - yes. If not - in most cases it's a wrong input, exceptions are the cases when we cannot distinguish either endpoint is unreachable (network issue) or wrong host.